azurerm/resource_arm_app_service_plan.go

```diff
@@ -0,0 +1,215 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"bytes"
+
+	"github.com/Azure/azure-sdk-for-go/arm/web"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppServicePlan() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServicePlanCreateUpdate,
+		Read:   resourceArmAppServicePlanRead,
+		Update: resourceArmAppServicePlanCreateUpdate,
+		Delete: resourceArmAppServicePlanDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": locationSchema(),
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"size": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAzureRMAppServicePlanSkuHash,
+			},
+			"maximum_number_of_workers": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppServicePlanCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	AppServicePlanClient := client.appServicePlansClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM App Service Plan creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	sku := expandAzureRmAppServicePlanSku(d)
+
+	properties := web.AppServicePlanProperties{}
+	if v, ok := d.GetOk("maximum_number_of_workers"); ok {
+		maximumNumberOfWorkers := v.(int32)
```

Comment:
could we pull this out of the schema as an int, and then casting the result to an int32? i.e.

```
maximumNumberOfWorkers = int32(v.(int))
```

(whilst perhaps an odd-looking comment/ask - this should guarantee compatibility going forward)